dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.cpp

```diff
@@ -195,6 +200,22 @@ namespace winrt::Microsoft::Windows::AppNotifications::Builder::implementation
         return *this;
     }
 
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddTextBox(hstring id)
+    {
+        ThrowIfMaxInputItemsExceeded();
+
+        m_textBoxList.push_back(wil::str_printf<std::wstring>(L"<input id='%ls' type='text'/>", id.c_str()));
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddTextBox(hstring id, hstring placeHolderText, hstring title)
+    {
+        ThrowIfMaxInputItemsExceeded();
+
```

Comment:
The only required param is id. We should prevent the placeholder/title from being added if empty, but the only time we should throw is if id is empty. 

Main example is developer may only want placeholder OR title instead of both.

Reply:
But the string formatting below goes and adds empty title andplaceholdertext as an attribution. Why have the attribute at all if they are empty?

Reply:
Is adding empty string a problem? Is that something that the xml schema would prohibit?

Reply:
If adding an empty string isn't illegal, then the work can be addressed when we get to it: https://microsoft.visualstudio.com/OS/_workitems/edit/40746502